libs/utils/src/celix_convert_utils.c

```diff
@@ -114,7 +114,7 @@ celix_version_t* celix_utils_convertStringToVersion(const char* val, const celix
         if (firstDot != NULL && lastDot != NULL && firstDot != lastDot) {
             char buf[64];
             char* valCopy = celix_utils_writeOrCreateString(buf, sizeof(buf), "%s", val);
-            char *trimmed = utils_stringTrim(valCopy);
+            char *trimmed = celix_utils_trimInPlace(valCopy);
```

Comment:
nice, I think this also fixes an bug. Because in the old situation the trimmed copy got freed in `celix_utils_freeStringIfNotEqual`, but if `celix_utils_writeOrCreateString` used a allocation this would leak.